src/driver/driver.hpp

```diff
@@ -85,10 +85,8 @@ template <typename T, class... Args>
 TaskListStatus ConstructAndExecuteBlockTasks(T *driver, Args... args) {
   int nmb = driver->pmesh->GetNumMeshBlocksThisRank(Globals::my_rank);
   std::vector<TaskList> task_lists;
-  MeshBlock *pmb = driver->pmesh->pblock;
-  while (pmb != nullptr) {
-    task_lists.push_back(driver->MakeTaskList(pmb, std::forward<Args>(args)...));
-    pmb = pmb->next;
+  for (auto &mb : driver->pmesh->block_list) {
+    task_lists.push_back(driver->MakeTaskList(&mb, std::forward<Args>(args)...));
```

Comment:
This needs is's own issue, it looks like the  MakeTaskList method does not necessarily exist in the driver class. This should probably be made a pure virtual method which forces users writing the driver to override it. Unless, that was some reason why the MakeTaskList should be optional.

Reply:
I recall that when @jdolence designed the driver, there was some reason `MakeTaskList` wasn't required by the driver. One reason is that not all drivers have a `step` iteration that should have a task list... for example the `pi` problem. Maybe @jdolence can comment more.

Reply:
I'm still wondering if it might be better practice to have a pure virtual method, that way you enforce the correct prototype, you can always have an empty method if it is not needed. I don't know the answer though...